Add overload name to JIT prim operators.

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29664 Add missing operators for PyText model.
* **#29656 Add overload name to JIT prim operators.**
* #29635 OPN ops TupleConstruct/Unpack and format.

Differential Revision: [D18499600](https://our.internmc.facebook.com/intern/diff/D18499600)

Commit:
Add overload name to JIT prim operators.

Commit:
Update on "Add overload name to JIT prim operators."

Commit:
Update on "Add overload name to JIT prim operators."

Commit:
Update on "Add overload name to JIT prim operators."

zdevito reviewed (approved):
Looks good, but the schema in the mobile registration need to be checked for correctness.

iseeyuan commented on torch/csrc/jit/mobile/register_mobile_ops.cpp:
```diff
@@ -315,4 +326,17 @@ static auto registry0 = torch::RegisterOperators().op(
   []() {
   })
+).op(
+  "_aten::append.Tensor(Tensor self) -> void",
```
Yep, schema in this file may be wrong. @smessmer Just curious: if we are using names only, would the schema be necessary in mobile? It looks like schema is required for boxed registration, but it's not checked in the registration itself. Could you confirm if a correct schema is required for registrations in this file. If it is, how to know what's the correct schema?

smessmer replied:
if you register multiple ops with the same operator name + overload name but mismatching schemas, the operator library will throw an error. This didn't happen here because `_aten::append.Tensor` is allowed to mismatch with `aten::append.Tensor`, they're different ops.

Commit:
Update on "Add overload name to JIT prim operators."

iseeyuan committed:
Update on "Add overload name to JIT prim operators."


Differential Revision: [D18499600](https://our.internmc.facebook.com/intern/diff/D18499600)

[ghstack-poisoned]

iseeyuan committed:
Update on "Add overload name to JIT prim operators."


Differential Revision: [D18499600](https://our.internmc.facebook.com/intern/diff/D18499600)

[ghstack-poisoned]

iseeyuan committed:
Update on "Add overload name to JIT prim operators."


Differential Revision: [D18499600](https://our.internmc.facebook.com/intern/diff/D18499600)

[ghstack-poisoned]

iseeyuan committed:
Update on "Add overload name to JIT prim operators."


Differential Revision: [D18499600](https://our.internmc.facebook.com/intern/diff/D18499600)

[ghstack-poisoned]

iseeyuan committed:
Update on "Add overload name to JIT prim operators."


Differential Revision: [D18499600](https://our.internmc.facebook.com/intern/diff/D18499600)

[ghstack-poisoned]

zdevito commented:
This is causing a test failure on master, I've started the unland to fix:
https://app.circleci.com/jobs/github/pytorch/pytorch/3626363
